src/loaddb/load_sa_loader.cpp

```diff
@@ -1417,33 +1419,159 @@ ldr_assign_class_id (DB_OBJECT *class_, int id)
 static DB_OBJECT *
 ldr_find_class (const char *class_name)
 {
-  LC_FIND_CLASSNAME find;
   DB_OBJECT *class_ = NULL;
-  char realname[SM_MAX_IDENTIFIER_LENGTH];
+  LC_FIND_CLASSNAME found = LC_CLASSNAME_EXIST;
+  char realname[SM_MAX_IDENTIFIER_LENGTH] = { '\0' };
 
   /* Check for internal error */
-  if (class_name == NULL)
+  if (class_name == NULL || class_name[0] == '\0')
     {
       er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_GENERIC_ERROR, 0);
       display_error (0);
       return NULL;
     }
 
-  sm_downcase_name (class_name, realname, SM_MAX_IDENTIFIER_LENGTH);
+  sm_user_specified_name (class_name, realname, SM_MAX_IDENTIFIER_LENGTH);
+
   ldr_Hint_class_names[0] = realname;
 
-  find =
-	  locator_lockhint_classes (1, ldr_Hint_class_names, ldr_Hint_locks, ldr_Hint_subclasses, ldr_Hint_flags, 1,
+  found = locator_lockhint_classes (1, ldr_Hint_class_names, ldr_Hint_locks, ldr_Hint_subclasses, ldr_Hint_flags, 1,
 				    NULL_LOCK);
-
-  if (find == LC_CLASSNAME_EXIST)
+  if (found == LC_CLASSNAME_EXIST)
     {
       class_ = db_find_class (class_name);
+
+      ldr_Hint_class_names[0] = NULL;
+
+      return class_;
+    }
+
+  /* This is the case when the loaddb utility is executed with the --no-user-specified-name option as the dba user. */
+  if (db_get_client_type() == DB_CLIENT_TYPE_ADMIN_UTILITY && prm_get_bool_value (PRM_ID_NO_USER_SPECIFIED_NAME))
+    {
+      char other_class_name[DB_MAX_IDENTIFIER_LENGTH] = { '\0' };
+
+      ldr_find_class_by_query (realname, other_class_name, DB_MAX_IDENTIFIER_LENGTH);
+      if (other_class_name[0] != '\0')
+	{
+	  ldr_Hint_class_names[0] = other_class_name;
+
+	  found = locator_lockhint_classes (1, ldr_Hint_class_names, ldr_Hint_locks, ldr_Hint_subclasses, ldr_Hint_flags, 1,
+					    NULL_LOCK);
+	  if (found == LC_CLASSNAME_EXIST)
+	    {
+	      class_ = db_find_class (other_class_name);
+
+	      ldr_Hint_class_names[0] = NULL;
+
+	      return class_;
+	    }
+	}
     }
 
   ldr_Hint_class_names[0] = NULL;
 
-  return (class_);
+  return class_;
+}
+
+static int
+ldr_find_class_by_query (const char *name, char *buf, size_t buf_size)
+{
+#define QUERY_BUF_SIZE 2048
+#define CT_CLASS_NAME "_db_class"
```

Comment:
Is there any need to define it if it is to be used only within this function?
How about using const variables?

Reply:
Remove #define CT_CLASS_NAME "_db_class" and add #include "transform.h" .